dask_expr/expr.py

```diff
@@ -814,6 +814,9 @@ def columns(self):
     def _meta(self):
         if is_dataframe_like(self.frame._meta):
             return super()._meta
+        # if we are not a DataFrame and have a scalar, we reduce to a scalar
+        if pd.api.types.is_scalar(self.operands[1]):
+            return 0
```

Comment:
I'd like to avoid anything too pandas specific here (otherwise, I'll probably end up needing to change it to get the cudf backend working). Also, can't the scalar type be something other than an `int`?

Maybe if `self.operand("columns")` is not a `list`, we just return `meta_nonempty(self.frame._meta).iloc[0]`?

Reply:
Yes definitely, forgot to commit the meta_nonemtpy part 

list is not sufficient, but we can create an explicit check